contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java

```diff
@@ -142,4 +145,203 @@ public boolean CryptUnprotectData(DATA_BLOB pDataIn, PointerByReference szDataDe
+	 *            of this value, is ignored by this function.
+	 * @param pName
+	 *            A pointer to the CERT_NAME_BLOB structure to be converted.
+	 * @param dwStrType
+	 *            This parameter specifies the format of the output string. This
+	 *            parameter also specifies other options for the contents of the
+	 *            string.
+	 * @param psz
+	 *            A pointer to a character buffer that receives the returned string.
+	 *            The size of this buffer is specified in the csz parameter.
+	 * @param csz
+	 *            The size, in characters, of the psz buffer. The size must include
+	 *            the terminating null character.
+	 * @return Returns the number of characters converted, including the terminating
+	 *         null character. If psz is NULL or csz is zero, returns the required
+	 *         size of the destination string.
+	 */
+	int CertNameToStr(int dwCertEncodingType, Pointer pName, int dwStrType, char[] psz, int csz);
+
+	/**
+	 * The CertVerifyCertificateChainPolicy function checks a certificate chain to
+	 * verify its validity, including its compliance with any specified validity
+	 * policy criteria.
+	 * 
+	 * @param pszPolicyOID
+	 *            Current predefined verify chain policy structures are listed in
+	 *            the following table.
+	 * @param pChainContext
+	 *            A pointer to a CERT_CHAIN_CONTEXT structure that contains a chain
+	 *            to be verified.
+	 * @param pPolicyPara
+	 *            A pointer to a CERT_CHAIN_POLICY_PARA structure that provides the
+	 *            policy verification criteria for the chain. The dwFlags member of
+	 *            that structure can be set to change the default policy checking
+	 *            behavior.
+	 * @param pPolicyStatus
+	 *            A pointer to a CERT_CHAIN_POLICY_STATUS structure where status
+	 *            information on the chain is returned. OID-specific extra status
+	 *            can be returned in the pvExtraPolicyStatus member of this
+	 *            structure.
+	 * @return The return value indicates whether the function was able to check for
+	 *         the policy, it does not indicate whether the policy check failed or
+	 *         passed.
+	 * 
+	 *         If the chain can be verified for the specified policy, TRUE is
+	 *         returned and the dwError member of the pPolicyStatus is updated. A
+	 *         dwError of 0 (ERROR_SUCCESS or S_OK) indicates the chain satisfies
+	 *         the specified policy.
+	 * 
+	 *         If the chain cannot be validated, the return value is TRUE and you
+	 *         need to verify the pPolicyStatus parameter for the actual error.
+	 * 
+	 *         A value of FALSE indicates that the function wasn't able to check for
+	 *         the policy.
+	 */
+	boolean CertVerifyCertificateChainPolicy(int pszPolicyOID, CERT_CHAIN_CONTEXT pChainContext,
+			CERT_CHAIN_POLICY_PARA pPolicyPara, CERT_CHAIN_POLICY_STATUS pPolicyStatus);
```

Comment:
`pszPolicyOID` is a string, not an int. Beware - this is a LPCSTR so must always be called with ascii type mapping!

Reply:
Since the methods in Crypt32 have params for both UNICODE and ASCII, would it be okay to keep the type mapper as the DEFAULT one and use 'com.sun.jna.platform.win32.WTypes.LPSTR' for  pszPolicyOID?

Reply:
You can do this, but then you need to fix it first. LPSTR is defined as:

```C
typedef char CHAR;
typedef /* [string] */  __RPC_string CHAR *LPSTR;
```

So it is an array of chars. The current definition in `WTypes.java` maps it to w_char. This should to it:

```java
    public static class LPSTR extends PointerType {
        public static class ByReference extends LPSTR implements
                Structure.ByReference {
        }

        public LPSTR() {
            super(Pointer.NULL);
        }

        public LPSTR(Pointer pointer) {
            super(pointer);
        }

        public LPSTR(String value) {
            this(new Memory(value.length() + 1L));
            this.setValue(value);
        }

        public void setValue(String value) {
            this.getPointer().setString(0, value);
        }

        public String getValue() {
            Pointer pointer = this.getPointer();
            String str = null;
            if (pointer != null)
                str = pointer.getString(0);

            return str;
        }

        @Override
        public String toString() {
            return this.getValue();
        }
    }
```

Please test this!

Reply:
Thanks. I start working on this and push out the changes i made for the other comments.

Reply:
Updated and tested this.